static/widgets/compiler-picker-popup.ts

```diff
@@ -0,0 +1,266 @@
+                e.preventDefault();
+                const elem = $(e.currentTarget);
+                elem.toggleClass('active');
+                const isa = unwrap(elem.attr('data-value'));
+                if (this.isaFilters.includes(isa)) {
+                    this.isaFilters = this.isaFilters.filter(v => v !== isa);
+                } else {
+                    this.isaFilters.push(isa);
+                }
+                this.fillCompilers();
+            });
+
+        // category filters
+        $(this.compilerTypes)
+            .find('.compiler-type')
+            .on('click', e => {
+                e.preventDefault();
+                const elem = $(e.currentTarget);
+                elem.toggleClass('active');
+                const category = unwrap(elem.attr('data-value'));
+                if (this.categoryFilters.includes(category)) {
+                    this.categoryFilters = this.categoryFilters.filter(v => v !== category);
+                } else {
+                    this.categoryFilters.push(category);
+                }
+                this.fillCompilers();
+            });
+    }
+
+    fillCompilers() {
+        const filteredIndices = this.searchResults
+            ? new Set(this.searchResults.items.map(item => item.id as number))
+            : undefined;
+        const filteredCompilers = this.options.filter((compiler, i) => {
+            if (this.isaFilters.length > 0) {
+                if (!this.isaFilters.includes(compiler.instructionSet || 'other')) {
+                    return false;
+                }
+            }
+            if (this.categoryFilters.length > 0) {
+                const categories = compiler.compilerCategories ?? ['other'];
+                if (intersection(this.categoryFilters, categories).length === 0) {
+                    return false;
+                }
+            }
+            if (filteredIndices) {
+                if (!filteredIndices.has(i)) {
+                    return false;
+                }
+            }
+            return true;
+        });
+        // figure out if there are any empty groups, these will be ignored
+        const groupCounts: Partial<Record<string, number>> = {};
+        for (const compiler of filteredCompilers) {
+            for (const group of compiler.$groups) {
+                groupCounts[group] = (groupCounts[group] ?? 0) + 1;
+            }
+        }
+        // add the compiler entries / group headers themselves
+        this.resultsContainer.empty();
+        this.favoritesContainer.empty();
+        const groupMap: Record<string, JQuery> = {};
+        for (const group of this.groups) {
+            if ((groupCounts[group.value] ?? 0) > 0 || group.value === CompilerPicker.favoriteGroupName) {
+                const group_elem = $(
+                    `
+                    <div class="group-wrapper">
+                        <div class="group">
+                            <div class="label">${group.label}</div>
```

Comment:
maybe escape the label? I'm not worried about security, just the usual `fish &amp; chips` that HTML has introduced into our lives ;)